### ISSUE TITLE
Application crashes if I do double insert with the same primary key

### ISSUE THREAD

**Issue by SSukhomlinov** [rbock/sqlpp11#303]
Example(mysql):

create table if not exists test
(
    player_id  int(11) unsigned not null primary key
);

auto db{ getConnection() };
account_db::Test test;
db(insert_into(test).set(test.playerId = 1));
db(insert_into(test).set(test.playerId = 1));

When i make insert first time(the Test table is clean), it just throws an exception as i'm expecting. But if i try to make same the second time(the table already contains data), it just crashes. 

Signal = SIGSEGV (Segmentation fault)

<unknown> 0x00007fd63d2accb0
__gxx_personality_v0 0x00007fd63d2ad5de
_Unwind_RaiseException 0x00007fd63d0ab04a
__cxa_throw 0x00007fd63d2adc9c
sqlpp::mysql::(anonymous namespace)::execute_statement(sqlpp::mysql::detail::connection_handle_t&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) 0x000055980fb6fac1
sqlpp::mysql::connection::insert_impl(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) 0x000055980fb7094c
sqlpp::mysql::connection::insert<sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, account_db::Test>, sqlpp::insert_list_t<void, sqlpp::assignment_t<sqlpp::column_t<account_db::Test, account_db::Test_::PlayerId>, sqlpp::unsigned_integral_operand> > > > connection.h:174
sqlpp::insert_t::_result_methods_t<sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, account_db::Test>, sqlpp::insert_list_t<void, sqlpp::assignment_t<sqlpp::column_t<account_db::Test, account_db::Test_::PlayerId>, sqlpp::unsigned_integral_operand> > > >::_run<sqlpp::mysql::connection> insert.h:73
sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, account_db::Test>, sqlpp::insert_list_t<void, sqlpp::assignment_t<sqlpp::column_t<account_db::Test, account_db::Test_::PlayerId>, sqlpp::unsigned_integral_operand> > >::_run<sqlpp::mysql::connection> statement.h:241
sqlpp::mysql::connection::_run<sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, account_db::Test>, sqlpp::insert_list_t<void, sqlpp::assignment_t<sqlpp::column_t<account_db::Test, account_db::Test_::PlayerId>, sqlpp::unsigned_integral_operand> > > > connection.h:257
sqlpp::mysql::connection::operator()<sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, account_db::Test>, sqlpp::insert_list_t<void, sqlpp::assignment_t<sqlpp::column_t<account_db::Test, account_db::Test_::PlayerId>, sqlpp::unsigned_integral_operand> > > > connection.h:266
account::api::AccountAPI::changeEmail EmailAPI.cpp:113
account::api::AccountAPI::onRequest AccountAPI.cpp:177
Pistache::Http::Handler::onInput(char const*, unsigned long, std::shared_ptr<Pistache::Tcp::Peer> const&) 0x00007fd63e1937dd
Pistache::Tcp::Transport::handleIncoming(std::shared_ptr<Pistache::Tcp::Peer> const&) 0x00007fd63e1c0564
Pistache::Tcp::Transport::onReady(Pistache::Aio::FdSet const&) 0x00007fd63e1c3af1
Pistache::Aio::SyncImpl::runOnce() 0x00007fd63e1bc2b5
std::thread::_State_impl<std::thread::_Invoker<std::tuple<Pistache::Aio::AsyncImpl::Worker::run()::{lambda()#1}> > >::_M_run() 0x00007fd63e1b9e9a
<unknown> 0x00007fd63d2d9f74
start_thread 0x00007fd63d434669
clone 0x00007fd63cfcb323

**Comment by rbock** [comment]
Hi,

the third line of the output starts with `_Unwind_RaiseException`. This seems to indicate that an exception was thrown but not caught? Can you post an [mvce](https://stackoverflow.com/help/minimal-reproducible-example), please?

Thanks,

Roland

**Comment by SSukhomlinov** [comment]
Hello. Thank you for your fast answer.

We tried to make mvce as you asked, but unfortunately we couldn't reproduce
this crash in our test project, although in our work project it is easy
reproducible. We have repeated everething like in our project, but without
success. There is a small defference between test project and work project.
The work project is an API-server over Pistache(C++ REST library). So we
will try to catch this crash with new mvce using Pistache. As soon as we'll
have any result or more information we'll get you to know.

Thanks a lot.

пт, 29 нояб. 2019 г. в 16:10, Roland Bock <notifications@github.com>:
